rpc/rpc-transport/socket/src/socket.c

```diff
@@ -4034,9 +4034,9 @@ ssl_setup_connection_params(rpc_transport_t *this)
     char *optstr = NULL;
     static int session_id = 1;
     int32_t cert_depth = DEFAULT_VERIFY_DEPTH;
-    char *cipher_list = DEFAULT_CIPHER_LIST;
-    char *dh_param = DEFAULT_DH_PARAM;
-    char *ec_curve = DEFAULT_EC_CURVE;
+    static char *cipher_list = DEFAULT_CIPHER_LIST;
+    static char *dh_param = DEFAULT_DH_PARAM;
+    static char *ec_curve = DEFAULT_EC_CURVE;
```

Comment:
I don't understand the reason to make these variables static. I don't see any benefit because the code is the same and they are always initialized, but it introduces a potential concurrency problem if more than one thread is running this function at the same time.

Reply:
OK, I can revert (or perhaps make const?)